core/src/main/java/org/alfresco/httpclient/HttpClientConfig.java

```diff
@@ -0,0 +1,215 @@
+
+    public void setProperties(Properties properties)
+    {
+        this.properties = properties;
+    }
+
+    public void setServiceName(String serviceName)
+    {
+        this.serviceName = serviceName;
+    }
+
+    public void setSslEncryptionParameters(SSLEncryptionParameters sslEncryptionParameters)
+    {
+        this.sslEncryptionParameters = sslEncryptionParameters;
+    }
+
+    public void setKeyResourceLoader(KeyResourceLoader keyResourceLoader)
+    {
+        this.keyResourceLoader = keyResourceLoader;
+    }
+
+    public AlfrescoKeyStore getKeyStore()
+    {
+        return keyStore;
+    }
+
+    public AlfrescoKeyStore getTrustStore()
+    {
+        return trustStore;
+    }
+
+    public void init()
+    {
+        this.keyStore = new AlfrescoKeyStoreImpl(sslEncryptionParameters.getKeyStoreParameters(),  keyResourceLoader);
+        this.trustStore = new AlfrescoKeyStoreImpl(sslEncryptionParameters.getTrustStoreParameters(), keyResourceLoader);
+
+        config = retrieveConfig(serviceName);
+        checkUnsupportedProperties(config);
+    }
+
+    /**
+     * Method used for retrieving HttpClient config from Global Properties
+     * @param serviceName name of used service
+     * @return map of properties
+     */
+    private Map<String, String> retrieveConfig(String serviceName)
+    {
+        return properties.keySet().stream()
+                .filter(key -> key instanceof String)
+                .map(Object::toString)
+                .filter(key -> key.startsWith(HTTPCLIENT_CONFIG + serviceName))
+                .collect(Collectors.toMap(
+                        key -> key.replace(HTTPCLIENT_CONFIG + serviceName + ".", ""),
+                        key -> properties.getProperty(key, null)));
+    }
+
+    private void checkUnsupportedProperties(Map<String, String> config)
+    {
+        config.keySet().stream()
+              .filter(propertyName -> !HttpClientProperties.isPropertyNameSupported(propertyName))
+              .forEach(propertyName -> LOGGER.warn(String.format("For service [%s], an unsupported property [%s] is set", serviceName, propertyName)));
+    }
+
+    private Integer getIntegerProperty(HttpClientProperties property)
+    {
+        return Integer.parseInt(extractValueFromConfig(property));
+    }
+
+    private Boolean getBooleanProperty(HttpClientProperties property)
+    {
+        return Boolean.parseBoolean(extractValueFromConfig(property));
+    }
+
+    private String extractValueFromConfig(HttpClientProperties property)
+    {
+        String keyValue;
+        if(property.isRequired)
+        {
+            keyValue = Optional.ofNullable(config.get(property.name)).orElseThrow(() -> {
+                String msg = String.format("Required property: '%s' is empty.", property.name);
+                LOGGER.error(msg);
+                throw new HttpClientException(msg);
+            });
+        } else {
+            keyValue = config.getOrDefault(property.name, "0");
```

Comment:
What about different property value types? Shouldn't it return null and the processing of a null return be done in the getInteger/getBoolean, or possibly in the future getStringProperty?

Is 0 parsed to false with parseBoolean?